paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/RepositoryLoader.kt

```diff
@@ -0,0 +1,1143 @@
+          handleParsingError(file, e)
+        }
+      }
+    }
+
+    // AAPT2 treats all styleable resources as public.
+    // See https://android.googlesource.com/platform/frameworks/base/+/master/tools/aapt2/ResourceParser.cpp#1539
+    resourceValue.namespaceResolver = namespaceResolver
+    return BasicResourceItem(
+      STYLEABLE,
+      name,
+      ResourceVisibility.PUBLIC,
+      configuration,
+      resourceValue
+    )
+  }
+
+  /**
+   * Adds attr definitions from [.myAttrs], and attr definition candidates from [.myAttrCandidates]
+   * if they don't match the attr definitions present in [.myAttrs].
+   */
+  private fun processAttrsAndStyleables() {
+    // for (attr in myAttrs.values()) {
+    //   addAttrWithAdjustedFormats(attr)
+    // }
+    // for (attr in myAttrCandidates.values()) {
+    //   val attrs: List<BasicAttrResourceItem> = myAttrs.get(attr.getName())
+    //   val i: Int = findResourceWithSameNameAndConfiguration(attr, attrs)
+    //   if (i < 0) {
+    //     addAttrWithAdjustedFormats(attr)
+    //   }
+    // }
+    //
+    // // Resolve attribute references where it can be done without loosing any data to reduce resource memory footprint.
+    // for (styleable in myStyleables.values()) {
+    //   addResourceItem(resolveAttrReferences(styleable))
+    // }
+  }
+
+  @Throws(IOException::class, XmlPullParserException::class)
+  private fun createFileReferenceItem(
+    type: ResourceType,
+    name: String,
+    configuration: RepositoryConfiguration
+  ): BasicResourceItem {
+    val namespaceResolver = parser.namespaceResolver
+    var text = textExtractor.extractText(parser, false).trim()
+    if (text.isNotEmpty() && !text.startsWith(PREFIX_RESOURCE_REF) && !text.startsWith(
+        PREFIX_THEME_REF
+      )
+    ) {
+      text = text.replace('/', File.separatorChar)
+    }
+    val visibility = getVisibility(type, name)
+    val resourceValue = ResourceValueImpl(namespace, type, name, text, null)
+    resourceValue.namespaceResolver = namespaceResolver
+    return BasicResourceItem(type, name, visibility, configuration, resourceValue)
+  }
+
+  @Throws(XmlSyntaxException::class)
+  private fun getResourceType(
+    tagName: String,
+    file: PathString
+  ): ResourceType? {
+    var type = ResourceType.fromXmlTagName(tagName)
+    if (type == null) {
+      if (TAG_EAT_COMMENT == tagName || TAG_SKIP == tagName) {
+        return null
+      }
+      if (tagName == TAG_ITEM) {
+        val typeAttr = parser.getAttributeValue(null, ATTR_TYPE)
+        if (typeAttr != null) {
+          type = ResourceType.fromClassName(typeAttr)
+          if (type != null) {
+            return type
+          }
+
+          LOG.warning("Unrecognized type attribute \"" + typeAttr + "\" at " + getDisplayName(file) + " line " + parser.lineNumber)
+        }
+      } else {
+        LOG.warning("Unrecognized tag name \"" + tagName + "\" at " + getDisplayName(file) + " line " + parser.lineNumber)
```

Comment:
nit: convert all Java-style string concatentations to Kotlin-style string templates?